framework/OutStreams/OutStreamPlot.py

```diff
@@ -331,40 +331,40 @@ def __fillCoordinatesFromSource(self):
             outputIndexes = self.sourceData[pltindex].indexes if xsplit[1].lower() == 'output' else []
             if xsplit[2].strip() not in self.sourceData[pltindex].getVars(xsplit[1].lower())+outputIndexes:
               self.raiseAnError(IOError, 'Not found variable "'+ xsplit[2] + '" in "'+xsplit[1]+ '" of DataObject "'+ self.sourceData[pltindex].name+'"!')
-            self.xValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[xsplit[2]].values)
-            maxSize = len(self.xValues[pltindex][cnt][-1]) if len(self.xValues[pltindex][cnt][-1]) > maxSize else maxSize
+            self.xValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[xsplit[2]].values.astype(float, copy=False))
+            maxSize = self.xValues[pltindex][cnt][-1].size if self.xValues[pltindex][cnt][-1].size > maxSize else maxSize
           if self.yCoordinates :
             for i in range(len(self.yCoordinates [pltindex])):
               ysplit = self.__splitVariableNames('y', (pltindex, i))
               outputIndexes = self.sourceData[pltindex].indexes if ysplit[1].lower() == 'output' else []
               if ysplit[2].strip() not in self.sourceData[pltindex].getVars(ysplit[1].lower())+outputIndexes:
                 self.raiseAnError(IOError, 'Not found variable "'+ ysplit[2] + '" in "'+ysplit[1]+ '" of DataObject "'+ self.sourceData[pltindex].name+'"!')
-              self.yValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[ysplit[2]].values)
-              maxSize = len(self.yValues[pltindex][cnt][-1]) if len(self.yValues[pltindex][cnt][-1]) > maxSize else maxSize
+              self.yValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[ysplit[2]].values.astype(float, copy=False))
+              maxSize = self.yValues[pltindex][cnt][-1].size if self.yValues[pltindex][cnt][-1].size > maxSize else maxSize
           if self.zCoordinates  and self.dim > 2:
             for i in range(len(self.zCoordinates [pltindex])):
               zsplit = self.__splitVariableNames('z', (pltindex, i))
               outputIndexes = self.sourceData[pltindex].indexes if zsplit[1].lower() == 'output' else []
               if zsplit[2].strip() not in self.sourceData[pltindex].getVars(zsplit[1].lower())+outputIndexes:
                 self.raiseAnError(IOError, 'Not found variable "'+ zsplit[2] + '" in "'+zsplit[1]+ '" of DataObject "'+ self.sourceData[pltindex].name+'"!')
-              self.zValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[zsplit[2]].values)
-              maxSize = len(self.zValues[pltindex][cnt][-1]) if len(self.zValues[pltindex][cnt][-1]) > maxSize else maxSize
+              self.zValues[pltindex][cnt].append(dataSet.isel(False,RAVEN_sample_ID=cnt)[zsplit[2]].values.astype(float, copy=False))
+              maxSize = self.zValues[pltindex][cnt][-1].size if self.zValues[pltindex][cnt][-1].size > maxSize else maxSize
```

Comment:
```pltindex``` happens in many places. Seems we can fix it later.

Reply:
I fixed all of them (pltindex, xsplit, etc.)